_posts/2017-07-14-Rust-Roadmap-Update.md

```diff
@@ -0,0 +1,292 @@
+work retooling the compiler to work in an incremental fashion.
+Earlier this year, we advertised
+[the "beta" release of incremental on nightly builds][icbeta].  While
+the beta version sometimes achieved quite large speedups, we also
+found that the dependency tracking not as robust or effective as we
+would like. Therefore, we are now adopting a revised approach to
+incremental compilation, which we call the "red-green algorithm" (I
+prefer "the salsa algorithm", but that has never caught on). We expect
+to be switching over to this new system in next month or two, which
+should mean that we will see much better incremental performance
+shortly thereafter. If you're interested in helping with the
+transition to the new system, check out
+[the incremental compilation roadmap issue][rr4] or
+[the tracking issue for the red-green algorithm itself][42293]; you
+can also follow [this internals thread][soyouwantic], where we
+regularly post links to bugs that include mentoring instructions.
+
+Looking beyond incremental compilation, we've also been taking steps
+to optimize compilation time in other ways. Probably the most
+important step in that respect has been getting a new version of
+[the `perf.rust-lang.org` website][perf] up and running. The "perf"
+website tracks the effect of each and every PR on compilation
+performance, so that we can detect and correct regressions. In fact,
+the new website immediately led to some
+[major performance improvements][41469]. There is still lots of work
+that could be done to improve it, however, ranging from evaluating and
+improving our benchmark suite to improving the web interface; see the
+[tracking issue on this topic][42611] for more.
+
+[1.16]: https://blog.rust-lang.org/2017/03/16/Rust-1.16.html
+[rr4]: https://github.com/rust-lang/rust-roadmap/issues/4
+[icbeta]: https://internals.rust-lang.org/t/incremental-compilation-beta/4721
+[42293]: https://github.com/rust-lang/rust/issues/42293
+[soyouwantic]: https://internals.rust-lang.org/t/so-you-want-to-help-with-incremental-compilation/5313
+[evaluating and improving out benchmark suite]: https://internals.rust-lang.org/t/measuring-compiler-performance/4966
+[tracking the runtime of generated code]: https://github.com/rust-lang/rust/issues/31265
+[perf]: http://perf.rust-lang.org/
+[improving the web interface]: https://github.com/rust-lang-nursery/rustc-perf/issues
+[41469]: https://github.com/rust-lang/rust/pull/41469
+[42611]: https://github.com/rust-lang/rust/issues/42611
+
+### Rust should provide a solid, but basic IDE experience
+
+Since it first debuted at RustConf last year, the Rust Language
+Service (RLS) has been growing rapidly ([roadmap issue][rr6]). It now
+offers support for most basic IDE operations, such as "jump to
+definition" or "find all uses", as well as offering code completion
+(via [the racer project](https://github.com/racer-rust/racer)) and
+some refactorings. At this point, the focus is primarily on polish:
```

Comment:
I would remove the refactoring bit - we only support renaming and that is behind a flag because it is not super-reliable